tests/lib.rs

```diff
@@ -173,13 +274,18 @@ fn write_random_sized_out_of_sequence_writes_with_gaps_and_overlaps() {
     let mut se = SelfEncryptor::new(&mut my_storage, datamap::DataMap::None);
     for ind in index {
       let piece_size : usize = pieces[ind].len();
-      let offset : usize = ind * piece_size;
+      let offset : usize = offsets[ind];
       total_size = std::cmp::max(total_size, offset + piece_size);
       se.write(pieces[ind], offset as u64);
-
-
+      assert!(DATA_SIZE >= total_size as u64);
+      // let decrypted = se.read(offset as u64, piece_size as u64);
```

Comment:
This should be committed as a failing test if the test is OK. We can then allocate somebody to fix the bug.